.github/workflows/tpp-benchmark.yml

```diff
@@ -25,18 +27,20 @@ env:
 
 jobs:
   Check_LLVM:
-    if: ${{ github.event_name }} == "push" || \
-        (${{ github.event_name }} == "pull_request" && ${{ github.event.label.name == 'benchmark' }}) || \
-        (${{ github.event_name }} == "pull_request" && ${{ github.event.label.name == 'benchmark-all' }})
+    if: |
+      (github.event_name == 'push') || (github.event_name == 'workflow_dispatch') ||
+      (github.event_name == 'pull_request' && contains(github.event.pull_request.labels.*.name, 'benchmark')) ||
+      (github.event_name == 'pull_request' && contains(github.event.pull_request.labels.*.name, 'benchmark-full'))
     uses: ./.github/workflows/tpp-llvm.yml
     secrets: inherit
 
   TPP-MLIR-EMR-BASE:
     runs-on: pcl-tiergarten
-    if: ${{ github.event_name }} == "push" || \
-        ${{ inputs.RUN_EMR_BENCH }} == 1 || \
-        (${{ github.event_name }} == "pull_request" && ${{ github.event.label.name == 'benchmark' }}) || \
-        (${{ github.event_name }} == "pull_request" && ${{ github.event.label.name == 'benchmark-all' }})
+    if: |
+      (github.event_name == 'push') || 
```

Comment:
Won't this trigger on any push?

Reply:
Any push to the main branch. There's a restriction on the trigger.